src/CommandProcessing/CommandProcessing.cpp

```diff
@@ -0,0 +1,197 @@
+#include "CommandProcessing.h"
+
+/************************************************************ CommandProcessor **************************************************************/
+/**
+ * Default Constructor
+ */
+    CommandProcessor::CommandProcessor(){}
+/**
+ * Destructor
+ */
+    CommandProcessor::~CommandProcessor(){}
+/**
+ * Assignment Operator
+ */
+    CommandProcessor& CommandProcessor::operator=(const CommandProcessor& other){
+        this->commands = other.commands;
+        return *this;
+    }
+/**
+ * Copy Constructor
+ */
+    CommandProcessor::CommandProcessor(const CommandProcessor& other){
+        this->commands = other.commands;
+    }
+/**
+ * overide Stream insertion operator
+ */
+    ostream& operator<<(ostream& out, CommandProcessor* cp){
+        //haven't decided what to printout so TODO!!!
+        return out;
+    }
+/**
+ * gets a command from the console as a string
+*/
+    Command* CommandProcessor::readCommand(){
+        string commandstr;
+        cout << "Enter a command:\n";
+        cin >> commandstr;
+        Command* commandobj = new Command(commandstr);
+        return commandobj;
+    }
+/**
+ * stores commands internally in a collection of Command objects
+ */
+    void CommandProcessor::saveCommand(){
+        Command* commandobj = readCommand();
+        commands.push_back(commandobj);
+    }
+/**
+ * public method for other objects such as the GameEngine or the Player
+ */
+    Command CommandProcessor::getCommandForOtherObjects(){
+        if (!commands.empty()) {
+            //TODO
```

Comment:
what should be done here?:)

Reply:
oh i just wrote to do as a self note because am not yet done with the code. i just wanted to publish what i have to give everyone an idea of what am working on in case they need it for their work. i'll commit again once i finish